File: 绿盟全国赛决赛参考答案/Java/第一阶段/step1/Task.java
package step1;

public class Task {
	
	public void sort(Integer[] arr){
		/********* Begin *********/
        
        int swapped1 = 0;
        int swapped2 = 0;
        int i, temp, left = 0, right = arr.length - 1;
        while (left < right) {
            for (i = left; i < right; i++) {
                if (arr[i] > arr[i + 1]) {
                    temp = arr[i];
                    arr[i] = arr[i + 1];
                    arr[i + 1] = temp;
                    swapped1 = 1;
                    //打印过程	
                    for(int a:arr)
                        System.out.print(a + " ");				
                    System.out.println();
                }
            }
            right--;
            for (i = right; i > left; i--)
            {
                if (arr[i - 1] > arr[i])
                {
                    temp = arr[i];
                    arr[i] = arr[i - 1];
                    arr[i - 1] = temp;
                    swapped2 = 1;
                    //打印过程	
                    for(int a:arr)
                        System.out.print(a + " ");				
                    System.out.println();
                }
            }
            left++;
        }
        if(swapped1 == 0 && swapped2 == 0){
            for(int a:arr)
                System.out.print(a + " ");				
            System.out.println();
        }

		/********* End *********/
	}
}

File: 绿盟全国赛决赛参考答案/Python/第一阶段/step1/Task.py
# -*- coding: utf-8 -*-
class Task:

    def sort(self,xlist):
    #********* Begin *********#
        global swapped1
        global swapped2
        swapped1 = 0
        swapped2 = 0
        left = 0
        right = len(xlist) - 1
        while left < right:
            for i in range(left,right):
                if xlist[i] > xlist[i + 1]:
                    self.Swap(xlist, i, i + 1)
                    swapped1 = 1
                    print(' '.join(map(str, xlist)))
            right -= 1

            for i in range(right,left,-1):
                if xlist[i - 1] > xlist[i]:
                    self.Swap(xlist, i - 1, i)
                    swapped2 = 1
                    print(' '.join(map(str, xlist)))
            left += 1
        if swapped1 == 0 and swapped2 == 0:
            print(' '.join(map(str, xlist)))        
    
    def Swap(self,xlist,i,j):
        temp = xlist[i]
        xlist[i] = xlist[j]
        xlist[j] = temp
    #********* End *********#